torch/distributed/_tensor/device_mesh.py

```diff
@@ -227,6 +227,26 @@ def _get_or_create_default_group(self):
                     "DeviceMesh must include every process in WORLD, "
                     f"but WORLD_SIZE({world_size}) != mesh size({self.mesh.numel()})"
                 )
+
+            unique_mesh_values = self.mesh.unique(sorted=True)
+            if unique_mesh_values.numel() != self.mesh.numel():
+                raise RuntimeError(
+                    f"DeviceMesh cannot have duplicate values, but found {self.mesh.tolist()}"
+                )
+
+            # ranks in mesh must start from 0
+            if unique_mesh_values[0] != 0:
+                raise RuntimeError(
+                    "DeviceMesh ranks must start from 0, "
+                    f"but found min rank = {unique_mesh_values[0]}"
+                )
+
+            # mesh must be contiguous (i.e. from 0 to N-1)
+            if 2 * unique_mesh_values.sum().item() != world_size * (world_size - 1):
+                raise RuntimeError(
+                    f"DeviceMesh is expected to be contiguous, but found {self.mesh.tolist()}"
```

Comment:
nit: it might not be contiguous as always, i.e. [[0, 2], [1, 3]] Maybe change the error msg to sth like:

DeviceMesh should have all ranks of the world